### ISSUE TITLE
Invalid IDE0004 on cast in switch expression

### ISSUE THREAD

**Issue by drewnoakes** [dotnet/roslyn#42108]
**Version Used**: Version 16.6.0 Preview 2.0 (master) (private branch build with unrelated changes)

**Steps to Reproduce**:

```c#
enum E { A, B }

class C
{
    bool M(int i)
    {
        return (E)i switch
        //     ~~~ IDE0004 cast is redundant
        {
            E.A => true,
            _ => false
        };
    }
}
```

**Expected Behavior**:

Cast is not flagged as redundant.

**Actual Behavior**:

Cast is flagged as redundant. Removing it introduces CS0266 on `E.A` in switch case.

**Comment by sharwell** [comment]
Duplicate of #37953

**Comment by astrohart** [comment]
Me too just now, am having issues.
